cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -343,6 +349,11 @@ public Void call() throws Exception {
                                                   String metricsConfPath, @Nullable String logbackJarName) {
     Map<String, String> configs = new HashMap<>();
 
+    // Make Spark UI runs on random port. By default, Spark UI runs on port 4040 and it will do a sequential search
+    // of the next port if 4040 is already occupied. However, during the process, it unnecessarily logs big stacktrace
+    // as WARN, which pollute the logs a lot if there are concurrent Spark job running (e.g. a fork in Workflow).
+    configs.put("spark.ui.port", "0");
```

Comment:
Spark will log the port when it starts. Also, in distributed mode, the YARN application master UI will get proxy to the actual Spark WebUI, so most of the time user won't even bother what port it is running on (and besides in cluster, one shouldn't run on a fixed port).